### PR TITLE
returns min possible size vol

### PR DESCRIPTION
**Pull request by boyamurthy**
allowing to create volume with minimum supported size. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1305 |

# Checklist:

- [X] Have you run format,vet & lint checks against your submission?
- [X] Have you made sure that the code compiles?
- [X] Did you run the unit & integration tests successfully?
- [ ] Have you maintained at least 90% code coverage?
- [ ] Have you commented your code, particularly in hard-to-understand areas
- [ ] Have you done corresponding changes to the documentation
- [X] Did you run tests in a real Kubernetes cluster?
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Unit test with new changes. 

image

### CODE REVIEW

**Review by delldubey — APPROVED** [review]
lgtm

File: service/controller.go [hunk 1]
```diff
@@ -1501,9 +1501,9 @@ func (s *service) validateVolSize(ctx context.Context, cr *csi.CapacityRange, sy
 			"bad capacity: requested minimum size (%d bytes) is greater than the maximum available capacity (%d bytes)", minSizeBytes, maxAvailBytes)
 	}
 	if minSizeBytes < MinVolumeSizeBytes {
-		return 0, status.Errorf(
-			codes.OutOfRange,
-			"bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
+		log.Warningf("bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
```
Comment (shanmydell) [inline service/controller.go:1504]: Requested size (%d bytes) is less than the minimum volume size (%d bytes) supported by PowerMax..

File: service/controller.go [hunk 2]
```diff
@@ -1501,9 +1501,9 @@ func (s *service) validateVolSize(ctx context.Context, cr *csi.CapacityRange, sy
...
+		log.Warning("provisioning vol with minimum volume size...")
```
Comment (shanmydell) [inline service/controller.go:1505]: Proceeding with minimum volume size supported by PowerMax Array ...

**Review by shanmydell — APPROVED** [review]

File: service/controller.go (hunk 1 above)
Comment (boyamurthy) [reply service/controller.go]: updated.

File: service/controller.go (hunk 2 above)
Comment (boyamurthy) [reply service/controller.go]: updated.

**Review by gallacher — CHANGES_REQUESTED** [review]
Please document the testing that was performed to validate this fix.  Have unit tests been put in place to test various scenarios around volume size?  If not, please ensure they are in. Not just happy path.  Thanks.

**Comment by boyamurthy** [comment]
> Please document the testing that was performed to validate this fix. Have unit tests been put in place to test various scenarios around volume size? If not, please ensure they are in. Not just happy path. Thanks.

Hi @gallacher , added the unit tests to cover the various volume sizes and added the screen shot for the same. Thank you.

**Review by gallacher — APPROVED** [review]

**Review by nidtara — APPROVED** [review]
lgtm

**Pull request merged**